dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java

```diff
@@ -211,21 +213,62 @@ public final Response deletePortletForRole(@Context final HttpServletRequest req
 
     }
 
+    /**
+     * This endpoint returns a portlet's details given its id
+     * @param request
+     * @param portletId
+     * @return
+     */
+    @GET
+    @JSONP
+    @Path("/{portletId}")
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findPortlet(@Context final HttpServletRequest request,
+            @PathParam("portletId") final String portletId) {
+
+        try {
+            final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                    .requiredBackendUser(true)
+                    .requiredFrontendUser(false)
+                    .requestAndResponse(request, null)
+                    .rejectWhenNoUser(true)
+                    .init();
+
+            final User user = initData.getUser();
+
+            if (!APILocator.getLayoutAPI().doesUserHaveAccessToPortlet(portletId, user)) {
```

Comment:
you can handle this on the WebResource.InitBuilder
Also you do not need to do a try catch, Jersey take care for us about that